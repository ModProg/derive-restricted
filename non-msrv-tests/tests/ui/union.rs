use std::marker::PhantomData;

use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(Debug)]
union UnsupportedTrait<T> {
	a: u8,
	b: PhantomData<T>,
}

#[derive(DeriveWhere)]
#[derive_where(Clone)]
union MissingCopy<T> {
	a: u8,
	b: PhantomData<T>,
}

fn main() {}
