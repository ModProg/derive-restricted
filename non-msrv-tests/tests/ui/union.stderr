error: traits other then `Clone` and `Copy` aren't supported by unions
 --> tests/ui/union.rs:5:1
  |
5 | / union Test<T> {
6 | |     field: PhantomData<T>,
7 | | }
  | |_^
